socket/src/commonTest/kotlin/mqtt/socket/SocketTests.kt

```diff
@@ -17,165 +17,122 @@ const val clientCount = 100L
     }
 
+    @ExperimentalUnsignedTypes
     @Test
     fun nioBlockingConnectDisconnectStressOpenConnections() = block {
-        val serverSocket = { asyncServerSocket() }
+        val serverSocket = { TCPServer("localhost", 0u, ServerProcessTest(ServerAction.CONNECT_DISCONNECT)) }
         val clientSocket = { clientSocket(true) }
-        val serverLaunched = launchServer(this, serverSocket)
-        stressTestOpenConnections(this, serverLaunched, serverSocket, clientSocket)
+        stressTestOpenConnections(serverSocket, clientSocket)
     }
 
-    private suspend fun launchServer(
-        scope: CoroutineScope,
-        getServerSocket: () -> ServerSocket
-    ): ServerLaunched {
-        val serverSocket = getServerSocket()
-        val firstReceiveLock = Mutex(true)
-        val mutex = Mutex(true)
-        var serverClientSocket: ClientSocket? = null
-        serverSocket.bind()
-        val server = Server(serverSocket)
-        val launched = ServerLaunched(firstReceiveLock, server, serverSocket, serverClientSocket, mutex)
-        scope.launch {
-            server.listen().collect {
-                serverClientSocket = it
-                launched.serverClientSocket = it
-                firstReceiveLock.unlock()
-                if (++launched.count >= clientCount) {
-                    mutex.unlock()
-                    return@collect
-                }
-            }
-        }
-        return launched
-    }
-
-    data class ServerLaunched(
-        val firstMessageReceivedLock: Mutex,
-        val server: Server,
-        val serverSocket: ServerSocket,
-        var serverClientSocket: ClientSocket?,
-        val mutex: Mutex,
-        var count: Int = 0
-    )
-
-    val limits = object : BufferMemoryLimit {
+    @ExperimentalUnsignedTypes
+    private val limits = object : BufferMemoryLimit {
         override fun isTooLargeForMemory(size: UInt) = size > 1_000u
     }
 
+    @ExperimentalUnsignedTypes
     private suspend fun connectClientReadWriteDisconnect(
         scope: CoroutineScope,
-        getServerSocket: () -> ServerSocket, getClientSocket: () -> ClientToServerSocket
+        getServerSocket: () -> TCPServer, getClientSocket: () -> ClientToServerSocket
     ) {
         val expectedClientToServer = 4.toUShort()
         val expectedServerToClient = UInt.MAX_VALUE
         val clientWriteBuffer = allocateNewBuffer(10.toUInt(), limits)
         clientWriteBuffer.write(expectedClientToServer)
         val clientReadBuffer = allocateNewBuffer(10.toUInt(), limits)
-        val serverSocket = getServerSocket()
-        assertFalse(serverSocket.isOpen())
-        serverSocket.bind()
-        assertTrue(serverSocket.isOpen())
-        val port = serverSocket.port()!!
+        val server = launchServer(getServerSocket())
+
+        val port = server.getListenPort()
         val clientToServerSocket = getClientSocket()
-        assertFalse(clientToServerSocket.isOpen())
```

Comment:
curious to why we dropped this assertFalse

Reply:
done

Reply:
right after the getClientSocket() call, I am making a call to initiateClient(). In it I am checking for the open status. Anyway, I added the check too.